src/pages/search.js

```diff
@@ -7,12 +7,22 @@ import GlobalSearchField from "components/search/GlobalSearchField";
 import DetailedSearchResults from "components/search/detailed/DetailedSearchResults";
 import SEARCH_RESULTS_TABS from "components/search/detailed/tabs";
 import NavigationConstants from "common/NavigationConstants";
+import { useConfig } from "contexts/config";
 
 export default function Search() {
+    const [config] = useConfig();
     const router = useRouter();
+    const [defaultTab, setDefaultTab] = useState(null);
     const { searchTerm, filter, selectedTab, advancedDataQuery } =
         router?.query;
-    let tab = selectedTab || SEARCH_RESULTS_TABS.data;
+    useEffect(() => {
+        setDefaultTab(
+            config?.elasticEnabled
+                ? SEARCH_RESULTS_TABS.data
+                : SEARCH_RESULTS_TABS.apps
+        );
+    }, [config]);
+    let tab = selectedTab || defaultTab;
```

Comment:
This `defaultTab` state and effect could also be replaced with a `const`:
```suggestion
    const defaultTab = config?.elasticEnabled
        ? SEARCH_RESULTS_TABS.data
        : SEARCH_RESULTS_TABS.apps;
    let tab = selectedTab || defaultTab;
```